shell.go

```diff
@@ -20,11 +22,42 @@ import (
 	tar "github.com/whyrusleeping/tar-utils"
 )
 
+const (
+	DefaultPathName = ".ipfs"
```

Comment:
Maybe use github.com/mitchellh/go-homedir instead and make this `DefaultIpfsPath = "~/.ipfs"`. That might simplify NewLocalShell a bit.